cdap-security/src/main/java/co/cask/cdap/security/tools/KeyStores.java

```diff
@@ -125,9 +127,17 @@ private static X509Certificate getCertificate(String dn, KeyPair pair, int days,
     X509CertInfo info = new X509CertInfo();
     info.set(X509CertInfo.VALIDITY, interval);
     info.set(X509CertInfo.SERIAL_NUMBER, new CertificateSerialNumber(sn));
+    // In java 7, subject is of type CertificateSubjectName and issuer is of type CertificateIssuerName.
+    // These were changed to X500Name in Java8. So looking at the field type before setting them.
     // This certificate will be self signed, hence the subject and the issuer are same.
-    info.set(X509CertInfo.SUBJECT, new CertificateSubjectName(owner));
-    info.set(X509CertInfo.ISSUER, new CertificateIssuerName(owner));
+    Field subjectField = info.getClass().getDeclaredField("subject");
```

Comment:
It's better to catch the `NoSuchFieldException` instead of throwing it out, as it's implementation detail. On handling the exception, you can default to the Java 8 behavior by using string (the assuming is Java 8 would be more forward compatible that Java 7)